main.cpp

```diff
@@ -115,17 +135,20 @@ struct <#structname2#>
 
 int main()
 {
-    T <#name1#>( , );                                             //6
-    T <#name2#>( , );                                             //6
+    T t1(73, "Colin");                                             //6
+    T t2(71, "Lynn");                                             //6
     
-    <#structName1#> f;                                            //7
-    auto* smaller = f.compare( , );                              //8
-    std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    Comparer f;                                            //7
+    auto* smaller = f.compare(&t1, &t2);                              //8
+    if(smaller != nullptr)
+        std::cout << "the smaller one is << " << smaller->name << std::endl; //9
+    else
+        std::cout << "Warning: smaller is pointing to a nullptr which was returned from f.compare()." << std::endl;
```

Comment:
I have recently updated the instructions for this particular part of the project to the following:
```
 9) correct the cout that uses smaller's member variable with some safe pointer usage.
    the possible return values of the compare() function should give you a hint.
    Also: if smaller == nullptr, make the cout statement explain the reasons that `f.compare()` might return nullptr.  
    hint: There are multiple reasons
```
- [ ] revise